code/__HELPERS/unsorted.dm

```diff
@@ -2017,5 +2017,18 @@ GLOBAL_DATUM_INIT(dview_mob, /mob/dview, new)
 	set waitfor = FALSE
 	return call(source, proctype)(arglist(arguments))
 
+/proc/IsFrozen(var/atom/A)
+	if(istype(A, /mob/living))
```

Comment:
Instead of hard coding these conditions, I would use the `GLOB.frozen_atom_list` list we have. Anything that is frozen will be added to this list.

So instead you can do something like 
```js
/proc/IsFrozen(var/atom/A)
    if(A in GLOB.frozen_atom_list)
        return TRUE
    return FALSE
```

Reply:
I modified and tested it. I think I re-up'd it correctly.